### DISCUSSION TITLE
Why am I getting ReadRequests instead of PublishRequests when I subscribe?

### DISCUSSION THREAD

**Discussion by imaginaryBuddy** [FreeOpcUa/opcua-asyncio#1641] [Q&A]
Hi, previously, when I was using the deprecated opcua library, when I made a subscription, Wireshark shows publish requests and responses but now when I migrated to asyncua library, using the sync version, I am getting read requests and responses instead. 
Are there any reasons for this? And how do I change it back to publish requests and responses? 

Thank you.

**Comment by oroulet** [comment]
You are most probably not. Make a minimal example and check. Maybe share it here if you do not understand the output

**Reply by imaginaryBuddy** [reply]
hi @oroulet , I realised the "infinite" read requests was coming from a read method call, not the subscribe call. I've raised the issue here, with a minimal example: 

https://github.com/FreeOpcUa/opcua-asyncio/issues/1642

**Comment by schroeder-** [comment]
There is a background task that sends readrequests, to keep the session alive, forgot the intervall. Maybe that is confusing you.

**Reply by imaginaryBuddy** [reply]
hi @schroeder- do you know how to disable the background task that sends readrequests?
